js/local-storage.js

```diff
@@ -0,0 +1,44 @@
+// store form fields as single object in local storage
+function storeFormField(e) {
+  /* eslint-disable */
+  const field = e.target.dataset.field;
+  /* eslint-enable  */
+  // get the form fields
+  const fieldValue = document.getElementById(field).value;
+  // unserialize form object from local storage
+  const formFields = JSON.parse(localStorage.getItem('formObject'));
+  // update name field in formfields object
+  formFields[field] = fieldValue;
+  // serialize form object to local storage
+  localStorage.setItem('formObject', JSON.stringify(formFields));
+}
+
+// check if formObject is in local storage
+function checkFormObject() {
+  if (!localStorage.getItem('formObject')) {
+    // if not, create an empty object
+    const formObject = {
+      name: '',
+      email: '',
+      message: '',
+    };
+    // serialize the object to local storage
+    localStorage.setItem('formObject', JSON.stringify(formObject));
+  } else {
+    // if yes, unserialize the object from local storage
+    const formObject = JSON.parse(localStorage.getItem('formObject'));
+    // set the form fields to the values from the object
+    document.querySelector('#name').value = formObject.name;
+    document.querySelector('#email').value = formObject.email;
+    document.querySelector('#message').value = formObject.message;
+  }
+}
+
+checkFormObject();
+
+// trigger storeFormFields function on name change
```

Comment:
- [ ] Kindly make sure that when the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data. Right now when I reload your browser The fields are empty.
- You can view this loom video to understand better [here](https://www.loom.com/share/96d15f69a5174364be61b521cc82ab2b)

You can refer the below code and can use it for the completing the requirement-:
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```